frame/lending/src/tests.rs

```diff
@@ -1506,26 +1520,25 @@ fn create_market<const NORMALIZED_PRICE: u128>(
 		currency_pair: CurrencyPair::new(collateral_asset.id(), borrow_asset.id()),
 	};
 
-	Lending::create_market(Origin::signed(manager), config.try_into_validated().unwrap()).unwrap();
-
-	let system_events = System::events();
-	if let Some(EventRecord {
-		event:
-			Event::Lending(crate::Event::<Runtime>::MarketCreated {
-				market_id,
-				vault_id,
-				manager: _,
-				currency_pair: _,
-			}),
-		..
-	}) = system_events.last()
-	{
-		(*market_id, *vault_id)
+	crate::Pallet::<R>::create_market(
+		SystemOrigin::<R>::signed(manager),
+		config.try_into_validated().unwrap(),
+	)
+	.unwrap();
+	let system_events = frame_system::Pallet::<R>::events();
+	let last_system_event = system_events.last().expect("There are no events in System::events()");
+	let pallet_event: crate::Event<R> = last_system_event
+		.event
+		.clone()
+		.try_into()
+		.expect("Market was not created due to System::Event => crate::Event conversion error");
+	if let crate::Event::<R>::MarketCreated { market_id, vault_id, .. } = pallet_event {
+		(market_id, vault_id)
 	} else {
 		panic!(
-			"System::events() did not contain the market creation event. Found {:#?}",
+			"There is no market creation event in System::events(). Found: {:#?}",
 			system_events
-		)
+		);
 	}
```

Comment:
I think assert_extrinsic_event could be used here, I wrote this test before I wrote that function and never changed the test to use it

Reply:
@benluelo As far as I understand, here we use the latest system event to extract market_id and vault_id, since dispatchable function do not return such information. assert_extrincis_event would not help us in this case.

Reply:
Yep, totally forgot we returned those values. Disregard haha